File: sol/solution/0001-0100/0012. Integer to Roman/Solution.rb
# @param {Integer} num
# @return {String}
def int_to_roman(num)
    val = [[1000, 'M'], [900, 'CM'], [500, 'D'], [400, 'CD'], [100, 'C'], [90, 'XC'], [50, 'L'], [40, 'XL'], [10, 'X'], [9, 'IX'], [5, 'V'], [4, 'IV'], [1, 'I']]
    res = ''
    val.each do |v, s|
        while num >= v
            res += s
            num -= v
        end
    end
    res
end

File: sol/solution/0001-0100/0012. Integer to Roman/Solution.c
char *intToRoman(int num)
{
    static char res[20];
    int val[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    char *sy[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
    int i = 0, p = 0;
    while (num)
    {
        while (num >= val[i])
        {
            for (char *s = sy[i]; *s; s++)
                res[p++] = *s;
            num -= val[i];
        }
        i++;
    }
    res[p] = 0;
    return res;
}